src/object/srv_obj.c

```diff
@@ -1299,24 +1299,53 @@ daos_iod_recx_dup(daos_iod_t *iods, uint32_t iod_nr, daos_iod_t **iods_dup_ptr)
 	return 0;
 }
 
+static int
+get_agg_hae(struct obj_io_context *ioc)
+{
+	uint64_t        ec_agg_boundary = ioc->ioc_coc->sc_ec_agg_eph_boundary;
+	vos_cont_info_t cinfo;
+	int             rc;
+
+	if (ec_agg_boundary != 0) {
+		ioc->ioc_agg_hae = ec_agg_boundary;
+		return 0;
+	} else if (ioc->ioc_agg_hae != 0) {
+		return 0;
+	}
+
+	/* query vos aggregation hae */
+	rc = vos_cont_query(ioc->ioc_vos_coh, &cinfo);
+	if (rc) {
+		D_ERROR(DF_CONT " cont query failed, " DF_RC "\n",
+			DP_CONT(ioc->ioc_coc->sc_pool->spc_uuid, ioc->ioc_coc->sc_uuid), DP_RC(rc));
+		return rc;
+	}
+	ioc->ioc_agg_hae = cinfo.ci_hae;
```

Comment:
This looks not quite right to me, but I think the delay 10 minutes to trigger rebuild on restart should solve most race issues for this moment.

Reply:
agreed, we can refine it maybe store it in RDB later.